File: app/src/main/java/com/wapchief/imitationlol/ScrollableFragment.java
package com.wapchief.imitationlol;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.ScrollView;

import com.bartoszlipinski.recyclerviewheader2.RecyclerViewHeader;
import com.cpoopc.scrollablelayoutlib.ScrollableHelper;

import java.util.ArrayList;
import java.util.List;

import butterknife.ButterKnife;
import butterknife.InjectView;

/**
 * Created by apple on 2017/6/20.
 */

public class ScrollableFragment extends Fragment implements ScrollableHelper.ScrollableContainer {

    @InjectView(R.id.header)
    RecyclerViewHeader header;
    @InjectView(R.id.recyclerView)
    RecyclerView recyclerView;
    @InjectView(R.id.scrollView)
    ScrollView scrollView;
    private View view;
    List<MainBean> data = new ArrayList<>();
    MainBean bean;
    FragmentAdapter adapter;
    RelativeLayout relativeLayout;
    ScrollableActivity activity;
    @Override
    public View getScrollableView() {
        return view;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // TODO: inflate a fragment view
        view = View.inflate(getActivity(), R.layout.fragment_listview, null);
        ButterKnife.inject(this, view);
        initView();
        activity= (ScrollableActivity) getActivity();
        return view;
    }

    private void initView() {
        initAdapter();
        initData();
        initonScroll();
    }

    private void initonScroll() {
        scrollView.setOnScrollChangeListener(new View.OnScrollChangeListener() {
            @Override
            public void onScrollChange(View view, int i, int i1, int i2, int i3) {
                Log.e("i======", "i:" + i + ",i1:" + i1 + ",i2:" + i2 + ",i3:" + i3);
                if (i1>300) {
                    activity.getonScroll(i1);
                }
            }
        });
        //这里获取不到recycler的监听事件，因为ScrollView冲突问题，在布局将滚动焦点交给了ScrollView处理
        recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
            @Override
            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
                super.onScrollStateChanged(recyclerView, newState);
            }

            @Override
            public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
                super.onScrolled(recyclerView, dx, dy);
                Log.e("d======", dx+","+dy);

            }
        });
    }

    private void initData() {

        for (int i = 0; i < 10; i++) {
            data.add(i, bean);
        }
        adapter.notifyDataSetChanged();
    }

    private void initAdapter() {
        View headerView = View.inflate(getContext(), R.layout.view_header, null);
        relativeLayout = (RelativeLayout) headerView.findViewById(R.id.header_title);

        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
        recyclerView.setLayoutManager(layoutManager);
//        header.addView(headerView);
//        header.attachTo(recyclerView);
        adapter = new FragmentAdapter(data, getActivity());
        //分割线
        recyclerView.addItemDecoration(new DividerItemDecoration(getActivity(), DividerItemDecoration.VERTICAL));
        recyclerView.setAdapter(adapter);
    }


    @Override
    public void onDestroyView() {
        super.onDestroyView();
        ButterKnife.reset(this);
    }
}

File: app/src/main/java/com/wapchief/imitationlol/ScrollableFragment.kt
package com.wapchief.imitationlol

import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cpoopc.scrollablelayoutlib.ScrollableHelper.ScrollableContainer
import kotlinx.android.synthetic.main.fragment_listview.*
import java.util.*

/**
 * Created by apple on 2017/6/20.
 */
class ScrollableFragment : Fragment(), ScrollableContainer {

//    var view: View
    var data: MutableList<MainBean?> = ArrayList()
    var bean: MainBean? = null
    var adapter: FragmentAdapter? = null
    var relativeLayout: RelativeLayout? = null
    var activity: ScrollableActivity? = null
    override fun getScrollableView(): View {
        return view!!
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        // TODO: inflate a fragment view
        activity = getActivity() as ScrollableActivity?
        return View.inflate(getActivity(), R.layout.fragment_listview, null)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView()
    }

    private fun initView() {
        initAdapter()
        initData()
        initonScroll()
    }

    private fun initonScroll() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            scrollView!!.setOnScrollChangeListener { view, i, i1, i2, i3 ->
                Log.e("i======", "i:$i,i1:$i1,i2:$i2,i3:$i3")
                if (i1 > 300) {
                    activity!!.getonScroll(i1)
                }
            }
        }
        //这里获取不到recycler的监听事件，因为ScrollView冲突问题，在布局将滚动焦点交给了ScrollView处理
        recyclerViewF!!.addOnScrollListener(object : RecyclerView.OnScrollListener() {
            override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
                super.onScrollStateChanged(recyclerView, newState)
            }

            override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
                super.onScrolled(recyclerView, dx, dy)
                Log.e("d======", "$dx,$dy")
            }
        })
    }

    private fun initData() {
        for (i in 0..9) {
            data.add(i,bean)
        }
        adapter!!.notifyDataSetChanged()
    }

    private fun initAdapter() {
        val headerView = View.inflate(context, R.layout.view_header, null)
        relativeLayout = headerView.findViewById<View>(R.id.header_title) as RelativeLayout
        val layoutManager = LinearLayoutManager(getActivity())
        recyclerViewF!!.layoutManager = layoutManager
        header.addView(headerView);
        header.attachTo(recyclerViewF);
        adapter = getActivity()?.let { FragmentAdapter(data, it) }
        //分割线
        recyclerViewF!!.addItemDecoration(DividerItemDecoration(getActivity(), DividerItemDecoration.VERTICAL))
        recyclerViewF!!.adapter = adapter
    }

    override fun onDestroyView() {
        super.onDestroyView()
    }
}